### ISSUE TITLE
Yet another stupid poem #1: Neznayka's method

### ISSUE THREAD

**Issue by schas002** [NaPoGenMo/NaPoGenMo2017#3]
This is my first entry into a *GenMo; so don't judge strictly please. (ovo)

[Source](https://gist.github.com/108f75319adb8c6d6e18df5f267f4bb3) in [Frink](https://frinklang.com/).

Basically: It picks an item in the dictionary file and selects others in the same file until the last `ENDING_THRESHOLD` characters match in both items. Spit them to output. Repeat.

```
item:     Initial commit
                     |||
ENDING_THRESHOLD: 3  ≠==
                      vv
item2:              init
For a threshold of 3, `Initial com*mit*` and `i*nit*` won't match...

item:     Initial commit
                     |||
ENDING_THRESHOLD: 3  ===
                     vvv
item2:              emit
...but `Initial com*mit*` and `e*mit*` will.

item:    Initial commit
                     ||
ENDING_THRESHOLD: 2  ==
                     vv
item2:             init
If the threshold is lowered, items which didn't match might.
```

**Comment by enkiv2** [comment]
Does this produce even-length lines with flat rhyme schemes consisting otherwise of random dictionary words? I'm not familiar with Frink & finding the logic a little hard to follow.

**Comment by schas002** [comment]
> Does this produce even-length lines...

Correct.

> ...with flat rhyme schemes...

Unsure what a flat rhyme scheme is...

> ...consisting otherwise of random dictionary words?

Yup.

**Comment by enkiv2** [comment]
By flat I mean A/A (as opposed to A/B/A or A/B/C/B)

**Comment by schas002** [comment]
Ah k :) Then yes, it does.

**Comment by schas002** [comment]
:D :D :D \o/ \o/ \o/

I finally got around to making a sample (see the next comment). It was made from a corpus of the 10 000 most used words from https://github.com/first20hours/google-10000-english (the no swear file, also excluding the first 30 items and every one-two letter items).

**Comment by schas002** [comment]
# Yet another stupid poem #1

For NaPoGenMo 2017

Authored by Andrew Zyabin

Last update: 2017-04-20

* * *

finding
passing
drama
panama


friendly
badly
cpu
cpu
workshops
develops


scenes
medicines
iowa
iowa
sphere
nowhere
poverty
party


hydrocodone
tone
education
identification
jail
jail
separation
dimension
wed
followed
russia
russia
horror
horror


december
chamber
pensions
calculations
leg
leg


handheld
springfield
corner
container
competitive
legislative
joining
mounting
cardiac
pontiac


scanning
examining
conservative
five
identify
identify
mem
mem


hon
python
checking
king
usr
usr
cake
snake
units
habits
serial
dial


resident
replacement
cialis
minneapolis
authorities
dictionaries
wages
mortgages
juice
notice


belt
felt
reflect
aspect
indians
hans


gains
wins

**Comment by schas002** [comment]
Wait, I accidentally pinged the "Resources" issue. To whoever sees the ping from any of the two sides, I'm sorry. :disappointed:

**Comment by MichaelPaulukonis** [comment]
Instead of going off of word-endings from the spelling, you might want to try translation to [Metaphone](https://en.wikipedia.org/wiki/Metaphone) _first_, and then go off of the resultant word-endings.

**Comment by schas002** [comment]
Hmm, I'd kind of like to see a YASP1 with a metaphoned dictionary. Just let me write up a translator...
